Make withExtendedLifetime, withUnsafePointer, etc inline(always)

stephentyrone opened the pull request.

stephentyrone committed:
Mark extendLifetime, withExtendedLifetime, withUnsafePointer, etc inline(always)

stephentyrone commented:
@swift-ci test

Catfish-Man reviewed (approved):
Surprised they weren't already, nice catch

stephentyrone committed:
Remove some CHECK lines for () type metadata.

When lifetime operations are inlined, we no longer need this, and so it doesn't necessarily get emitted?

stephentyrone commented:
@swift-ci test

aschwaighofer commented:
@swift-ci test
